pkgs/servers/sql/mariadb/default.nix

```diff
@@ -185,14 +180,6 @@ server = stdenv.mkDerivation (common // {
     chmod +x "$out"/bin/wsrep_sst_common
     rm "$out"/bin/{mysql_client_test,mysqltest}
     rm -r "$out"/data # Don't need testing data
-  '' + optionalString withoutClient ''
-    ${ # We don't build with GSSAPI on Darwin
-      optionalString (!stdenv.hostPlatform.isDarwin) ''
-        rm "$out"/lib/mysql/plugin/auth_gssapi_client.so
-      ''
-    }
```

Comment:
This code dont need remove.
https://github.com/NixOS/nixpkgs/pull/63355/commits/c8488b913a285b72bb9dc090f2903762b11ac9ce

Reply:
Yes it does - the whole thing is wrapped inside a `optionalString withoutClient` - so everything inside needs to go.

Reply:
Checked with this patch
```
diff --git a/pkgs/servers/sql/mariadb/default.nix b/pkgs/servers/sql/mariadb/default.nix
index 51181c4d2ae..da173b8b678 100644
--- a/pkgs/servers/sql/mariadb/default.nix
+++ b/pkgs/servers/sql/mariadb/default.nix
@@ -4,7 +4,6 @@
 , fixDarwinDylibNames, cctools, CoreServices, less
 , numactl # NUMA Support
 , withStorageMroonga ? true, kytea, msgpack, zeromq
-, withoutClient ? false
 }:

 with stdenv.lib;
@@ -149,9 +148,7 @@ server = stdenv.mkDerivation (common // {
     ++ optional stdenv.hostPlatform.isLinux linux-pam
     ++ optional (!stdenv.hostPlatform.isDarwin) mytopEnv;

-  patches = common.patches ++ [
-    ./cmake-without-client.patch
-  ] ++ optionals stdenv.hostPlatform.isDarwin [
+  patches = common.patches ++ optionals stdenv.hostPlatform.isDarwin [
     ./cmake-without-plugin-auth-pam.patch
   ];

@@ -170,8 +167,6 @@ server = stdenv.mkDerivation (common // {
     "-DWITH_NUMA=ON"
   ] ++ optional (!withStorageMroonga) [
     "-DWITHOUT_MROONGA=ON"
-  ] ++ optionals withoutClient [
-    "-DWITHOUT_CLIENT=ON"
   ] ++ optionals stdenv.hostPlatform.isDarwin [
     "-DWITHOUT_OQGRAPH=1"
     "-DWITHOUT_TOKUDB=1"
@@ -185,14 +180,11 @@ server = stdenv.mkDerivation (common // {
     chmod +x "$out"/bin/wsrep_sst_common
     rm "$out"/bin/{mysql_client_test,mysqltest}
     rm -r "$out"/data # Don't need testing data
-  '' + optionalString withoutClient ''
     ${ # We don't build with GSSAPI on Darwin
       optionalString (!stdenv.hostPlatform.isDarwin) ''
         rm "$out"/lib/mysql/plugin/auth_gssapi_client.so
       ''
     }
-    rm "$out"/lib/mysql/plugin/client_ed25519.so
-    rm "$out"/lib/{libmysqlclient${libExt},libmysqlclient_r${libExt}}
   '' + optionalString withStorageMroonga ''
     mv "$out"/share/{groonga,groonga-normalizer-mysql} "$out"/share/doc/mysql
   '' + optionalString (!stdenv.hostPlatform.isDarwin) ''

```

Reply:
`withoutClient` defaults to false, meaning, everything from https://github.com/NixOS/nixpkgs/pull/82649/files#diff-6967ff0d62d213fc6cd7c7d8207e3f11L188 till https://github.com/NixOS/nixpkgs/pull/82649/files#diff-6967ff0d62d213fc6cd7c7d8207e3f11L195 is always false and can be removed.

Your proposed patch would now still remove `$out"/lib/mysql/plugin/auth_gssapi_client.so` on non-darwin, even though we didn't do that before (as it was guarded inside  the `optionalString withoutClient`)

Reply:
> Your proposed patch would now still remove `$out"/lib/mysql/plugin/auth_gssapi_client.so` on non-darwin, even though we didn't do that before (as it was guarded inside the `optionalString withoutClient`)

https://github.com/NixOS/nixpkgs/commit/b8a6649797a73f75586c589a1b4c8197d9920c43#diff-6967ff0d62d213fc6cd7c7d8207e3f11R171-R177

Reply:
Sorry, dont need remove file `auth_gssapi_client.so`